packages/wonder-blocks-modal/src/components/flexible-panel.tsx

```diff
@@ -0,0 +1,211 @@
+import * as React from "react";
+import {PropsFor, View} from "@khanacademy/wonder-blocks-core";
+import type {StyleType} from "@khanacademy/wonder-blocks-core";
+
+import {focusStyles} from "@khanacademy/wonder-blocks-styles";
+import {StyleSheet} from "aphrodite";
+import {semanticColor} from "@khanacademy/wonder-blocks-tokens";
+import ModalContent from "./modal-content";
+import CloseButton from "./close-button";
+import theme from "../theme";
+
+export type BackgroundStyles = {
+    /**
+     * The background color of the panel. Defaults to semanticColor.surface.primary.
+     */
+    backgroundColor?: string;
+    /**
+     * The background image URL or gradient.
+     */
+    backgroundImage?: string;
+    /**
+     * How the background image should be repeated.
+     */
+    backgroundRepeat?: "repeat" | "no-repeat" | "repeat-x" | "repeat-y";
+    /**
+     * How the background image should be positioned.
+     */
+    backgroundPosition?: string;
+    /**
+     * How the background image should be sized.
+     */
+    backgroundSize?: string;
+    /**
+     * How the background image should fit within its container.
+     */
+    objectFit?: "fill" | "contain" | "cover" | "none" | "scale-down";
+};
+
+type RenderProps = {
+    title: React.ReactNode | string;
+};
+
+type Props = {
+    /**
+     * The main heading of the FlexiblePanel. Used to label the dialog.
+     */
+    title?: React.ReactNode | string;
+    /**
+     * Optional id reference for the main heading to label the parent dialog via aria-labelledby.
+     */
+    titleId?: string;
+    /**
+     * The main contents of the FlexiblePanel. All other parts of the panel
+     * are positioned around it.
+     */
+    content:
+        | React.ReactElement<PropsFor<typeof ModalContent>>
+        | ((slots: RenderProps) => React.ReactNode)
+        | React.ReactNode;
+    /**
+     * The optional background styles for the panel.
+     * If not provided, defaults to semanticColor.surface.primary background color.
+     */
+    backgroundStyles?: BackgroundStyles;
+    /**
+     * When true, the close button is shown; otherwise, the close button is not shown.
+     */
+    closeButtonVisible: boolean;
+    /**
+     * Should the contents of the panel become scrollable should they
+     * become too tall?
+     */
+    scrollOverflow: boolean;
```

Comment:
Is there a use case where we wouldn't want the panel to be scrollable if the contents become too tall? I'm wondering if this behaviour should always be true! (I see this is something in ModalPanel too, so it's pre-existing there!)